metrics/src/metrics.rs

```diff
@@ -136,8 +145,19 @@ impl MetricsAgent {
 
             let now = Instant::now();
             if now.duration_since(last_write_time) >= write_frequency && !points.is_empty() {
-                debug!("run: writing {} points", points.len());
-                writer.write(points);
+                debug!("run: attempting to write {} points", points.len());
+                if points.len() > max_points {
+                    warn!(
+                        "max submission rate of {} datapoints per second exceeded.  only the
+                    first {} of {} points will be submitted",
+                        rate_limit,
+                        max_points,
+                        points.len()
+                    );
+                    writer.write(points[..max_points].to_vec());
```

Comment:
Let's add a metric datapoint to signal that we're over the quota!  As a follow-up PR (or here if you prefer) then please add a new entry in the grafana dashboard to surface this datapoint (like panics or OOMs)

Reply:
Will do.

Reply:
Thanks oh might as well just generally report the number of points getting written out as well